pipeline/authn/authenticator_oauth2_client_credentials_cache_test.go

```diff
@@ -0,0 +1,96 @@
+package authn
+
+import (
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+
+	"github.com/ory/oathkeeper/driver/configuration"
+	"github.com/ory/viper"
+	"github.com/ory/x/logrusx"
+)
+
+func TestClientCredentialsCache(t *testing.T) {
+	viper.Reset()
+
+	ts := httptest.NewServer(httprouter.New())
+
+	viper.Set("authenticators.oauth2_client_credentials.config.token_url", ts.URL+"/oauth2/token")
+	viper.Set("authenticators.oauth2_client_credentials.config.cache.enabled", true)
+
+	logger := logrusx.New("", "")
+	c := configuration.NewViperProvider(logger)
+	a := NewAuthenticatorOAuth2ClientCredentials(c, logger)
+	assert.Equal(t, "oauth2_client_credentials", a.GetID())
+
+	config, err := a.Config(nil)
+	require.NoError(t, err)
+
+	t.Run("method=tokenToCache", func(t *testing.T) {
```

Comment:
Can you please add tests which ensure that we are not confusing credentials from e.g. config 1 and 2?